python/lsst/daf/butler/core/datastore.py

```diff
@@ -142,6 +142,11 @@ class Datastore(metaclass=ABCMeta):
     absolute path. Can be None if no defaults specified.
     """
 
+    containerKey = None
+    """Name of the key containing a list of subconfigurations that also
+    need to be merged with defaults and may use different Python types
+    (but the same config class). Assumed to be a list of configs."""
```

Comment:
`ConfigSubset` needs to know the container key used by each class, so it's definitely a datastore property and not a config property: we only have one DatastoreConfig but each datastore class has a different content in those configs: we could have DatastoreConfig subclasses but we don't do it that way at the moment). Saying that, I can remove this from here because `ConfigSubset` traps the case where the value hasn't been set for the class -- it was here mainly as a documentation aid for the property.